tst/schedulers/bayesopt/gpautograd/test_gp.py

```diff
@@ -55,11 +57,12 @@ def f(x):
 
     # to np.ndarray
     y_train_np_ndarray = anp.array(y_train)
-    x_train_np_ndarray = anp.array(x_train)
-    x_test_np_ndarray = anp.array(x_test)
+    x_train_np_ndarray = anp.array(x_train).reshape((-1, 1))
+    x_test_np_ndarray = anp.array(x_test).reshape((-1, 1))
 
     model = GaussianProcessRegression(kernel=Matern52(dimension=1))
-    model.fit(x_train_np_ndarray, y_train_np_ndarray)
+    data = {"features": x_train_np_ndarray, "targets": y_train_np_ndarray}
+    model.fit(data)
```

Comment:
what's the motivation for this API change?

Reply:
We have some models (in `gpautograd/learncurve`, where the data is not just represented by X and y, but by a dictionary. The whole interplay between likelihood and model (and optimizer) is based on this simplification.

I could in GaussianProcessRegression still also support the previous style, but this would be a bit confusing. All in all, this is an internal API.

What do you think?

Reply:
I could also call the new style `fit_from_dict`, and reserve `fit` for the old style?

Reply:
I don't know the code very well but is this extreme flexibility required there or do you always use the same args some of which are optional?

Reply:
Things are a bit messy for these models, because they support pre-computations of additional info depending on the data.

Reply:
I see. If they always take e.g. X and y, I would suggest to use fit(X, y, metadata) or similar. Otherwise I'm also fine with the current solution. Problem with passing dict is that it is very difficult to understand the API.